src/mui/list/InfiniteList.js

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { parse, stringify } from 'query-string';
+import { push as pushAction } from 'react-router-redux';
+import { Card, CardText } from 'material-ui/Card';
+import compose from 'recompose/compose';
+import { createSelector } from 'reselect';
+import inflection from 'inflection';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import autoprefixer from 'material-ui/utils/autoprefixer';
+import queryReducer, { SET_SORT, INC_PAGE, SET_PAGE, SET_FILTER, SORT_DESC } from '../../reducer/resource/list/queryReducer';
+import ViewTitle from '../layout/ViewTitle';
+import Title from '../layout/Title';
+import DefaultPagination from './Pagination';
+import DefaultActions from './Actions';
+import { crudGetList as crudGetListAction } from '../../actions/dataActions';
+import { changeListParams as changeListParamsAction } from '../../actions/listActions';
+import translate from '../../i18n/translate';
+import removeKey from '../../util/removeKey';
+import defaultTheme from '../defaultTheme';
+import InfiniteScroll from 'react-infinite-scroller';
+import LinearProgress from 'material-ui/LinearProgress';
+import { List } from './List';
+
+
+const styles = {
+    noResults: { padding: 20 },
+    header: {
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+};
+
+export class InfiniteList extends List {
+  constructor(props, state) {
+    super(props, state)
+    console.log(props);
+    console.log(state);
+  }
+
+  componentDidMount() {
+      this.updateData();
+      if (Object.keys(this.props.query).length > 0) {
+          this.props.query.page = 0;
+          this.props.changeListParams(this.props.resource, this.props.query);
+      }
+  }
+
+  getQuery() {
+      const query = Object.keys(this.props.query).length > 0 ?
+      Object.assign({}, this.props.query, {page: this.props.params.page}) : { ...this.props.params };
+      if (!query.sort) {
+          query.sort = this.props.sort.field;
+          query.order = this.props.sort.order;
+      }
+      if (!query.perPage) {
+          query.perPage = this.props.perPage;
+      }
+      return query;
+  }
+
+  updateData(query) {
```

Comment:
Is dup needed ?

Reply:
Yes because we set GrudGetList we diff parameters